inc/blocks/namespace.php

```diff
@@ -93,6 +95,49 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		return;
 	}
 
+	update_create_xb( $post_ID, $xbs, $post->post_title, '', '_xb_type' );
+
+}
+
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param array $instance The array of current widget settings.
+ * @param array $new_instance The array of new widget settings.
+ * @param array $old_instance The array of old widget settings.
+ * @param WP_Widget $widget The current widget instance.
+ * @return array $instance The current widget settings.
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	if ( ! isset( $instance['content'] ) ) {
+		return $instance;
+	}
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $instance['content'] );
+	$xbs = find_xbs( $blocks );
+
+	// Stop processing if no XBs as widgets.
+	if ( empty( $xbs ) ) {
+		return;
+	}
+
+	update_create_xb( '', $xbs, '', $widget->number, '_xb_is_widget' );
+
+	return $instance;
+}
+
+/**
+ * Update or create new XB's
+ *
+ * @param int $post_ID Post ID if applicable in use case, currently no post ID for widgets.
+ * @param array $xbs XB's on page or as widgets.
+ * @param string $post_title Page/Post title for XB Post.
+ * @param int $widget_number XB as a widget needs a number as the index doesn't increment.
+ * @param string $xb_meta_key Meta Key for updating the post_meta.
+ */
+function update_create_xb( $post_ID, $xbs, $post_title, $widget_number, $xb_meta_key ) {
```

Comment:
When you're writing a function try to consider which parameters are always required, which can have default values, and which are optional, then aim to put them in that order. You should also aim for some degree of making things generic when abstracting functionality. In this case I'd do something like:

```suggestion
function update_create_xb( array $xbs, string $default_title = '', ?int $item_id = null, string $context = 'post' ) {
```

Also note the type hinting - it forces a function to be used correctly and reduces the capacity for errors and creating bugs. The question mark before a type e.g. `?int` means that it's a nullable type, so the value can be `null` or an integer for example.